### PR TITLE
Update py-aiohttp to 3.7.4 and py-chartdet to 4.0

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/py-chardet/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyChardet(PythonPackage):
     homepage = "https://github.com/chardet/chardet"
     pypi = "chardet/chardet-3.0.4.tar.gz"
 
+    version('4.0.0', sha256='0d6f53a15db4120f2b08c94f11e7d93d2c911ee118b6b30a04ec3ee8310179fa')
     version('3.0.4', sha256='84ab92ed1c4d4f16916e05906b6b75a6c0fb5db821cc65e70cbd64a3e2a5eaae')
     version('3.0.2', sha256='4f7832e7c583348a9eddd927ee8514b3bf717c061f57b21dbe7697211454d9bb')
     version('2.3.0', sha256='e53e38b3a4afe6d1132de62b7400a4ac363452dc5dfcf8d88e8e0cce663c68aa')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-chardet/package.py:18]: It looks like `py-pytest-runner` is only required for 3.0.2, it gets skipped in all other versions. Also, can you add:
```python
depends_on('python@2.7:2.8,3.5:', when='@4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-aiohttp/package.py [hunk 2]
```diff
@@ -13,14 +13,19 @@ class PyAiohttp(PythonPackage):
...
 
+    version('3.7.4', sha256='d90ace66b55747e49531b13bf0a9c7ae9d1e14f315a56ea6b54c0dc1b6facd6e')
     version('3.6.2', sha256='4ad2d8393843ea0c7b50a93bf04fb1cf20b93c0a5e958de128efcd2e5f8adf80')
 
     depends_on('py-setuptools', type='build')
     depends_on('python@3.5.3:', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'), when='@3.7:')
     depends_on('py-attrs@17.3.0:', type=('build', 'run'))
     depends_on('py-chardet@2.0:3', type=('build', 'run'))
+    depends_on('py-chardet@2.0:4', type=('build', 'run'), when='@3.7.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aiohttp/package.py]: The previous condition was correct, it's <, not <=

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-chardet/package.py [hunk 3]
```diff
@@ -12,9 +12,11 @@ class PyChardet(PythonPackage):
...
 
     depends_on('py-setuptools', type=('build', 'run'))
-    depends_on('py-pytest-runner', type='build')
+    depends_on('py-pytest-runner', type='build', when='@:3')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-chardet/package.py]:
```suggestion
    depends_on('py-pytest-runner', type='build', when='@3.0.2')
```

File: var/spack/repos/builtin/packages/py-aiohttp/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-aiohttp/package.py]: Done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**